src/ipolates.F90

```diff
@@ -590,5 +591,215 @@ SUBROUTINE IPOLATES_grib2(IP,IPOPT,IGDTNUMI,IGDTMPLI,IGDTLENI, &
+  !! - (9):  NUMBER OF POINTS ALONG A MERIDIAN, OCTS 35-38
+  !! - (10): BASIC ANGLE OF INITIAL PRODUCTION DOMAIN, OCTETS 39-42
+  !! - (11): SUBDIVISIONS OF BASIC ANGLE, OCTETS 43-46
+  !! - (12): LATITUDE OF FIRST GRID POINT, OCTETS 47-50
+  !! - (13): LONGITUDE OF FIRST GRID POINT, OCTETS 51-54
+  !! - (14): RESOLUTION AND COMPONENT FLAGS, OCTET 55
+  !! - (15): LATITUDE OF LAST GRID POINT, OCTETS 56-59
+  !! - (16): LONGITUDE OF LAST GRID POINT, OCTETS 60-63
+  !! - (17): I-DIRECTION INCREMENT, OCTETS 64-67
+  !! - (18): J-DIRECTION INCREMENT, OCTETS 68-71
+  !! - (19): SCANNING MODE, OCTET 72
+  !! - (20): LATITUDE OF SOUTHERN POLE OF PROJECTION, OCTETS 73-76
+  !! - (21): LONGITUDE OF SOUTHERN POLE OF PROJECTION, OCTETS 77-80
+  !! - (22): ANGLE OF ROTATION OF PROJECTION, OCTS 81-84
+  !!
+  !! @param[in] igdtleni Number of elements of the grid definition
+  !! template array for the input grid. Corresponds to the gfld%igdtlen
+  !! component of the ncep g2 library gridmod data structure.
+  !!
+  !! @param[in] igdtnumo Grid definition template number for the output grid.
+  !! Corresponds to the gfld%igdtnum component of the
+  !! ncep g2 library gridmod data structure.
+  !! See "igdtnumi" for specific template definitions.
+  !! Note: igdtnumo<0 means interpolate to random station points.
+  !!
+  !! @param[in] igdtmplo Grid definition template array for the output grid.
+  !! Corresponds to the gfld%igdtmpl component of the ncep g2 library gridmod data structure.
+  !! See "igdtmpli" for definition of array elements.
+  !!
+  !! @param[in] igdtleno Number of elements of the grid definition template array for the output grid.  c
+  !! Corresponds to the gfld%igdtlen component of the ncep g2 library gridmod data structure.
+  !!
+  !! @param[in] mi    Skip number between input grid fields if km>1 or dimension of input grid fields if km=1.
+  !! @param[in] mo    Skip number between output grid fields if km>1 or dimension of output grid fields if km=1.
+  !! @param[in] km    Number of fields to interpolate.
+  !! @param[in] ibi   Input bitmap flags.
+  !! @param[in] li    Input bitmaps (if respective ibi(k)=1).
+  !! @param[in] gi    Input fields to interpolate.
+  !! @param[out] no Number of output points (only if kgdso(1)<0).
+  !! @param[out] rlat Output latitudes in degrees (if kgdso(1)<0).
+  !! @param[out] rlon Output longitudes in degrees (if kgdso(1)<0).
+  !! @param[out] ibo Output bitmap flags.
+  !! @param[out] lo  Output bitmaps (always output).
+  !! @param[out] go  Output fields interpolated.
+  !! @param[out] iret Return code.
+  !! - 0 Successful interpolation.
+  !! - 1 Unrecognized interpolation method.
+  !! - 2 Unrecognized input grid or no grid overlap.
+  !! - 3 Unrecognized output grid.
+  !! - 1x Invalid bicubic method parameters.
+  !! - 3x Invalid budget method parameters.
+  !! - 4x Invalid spectral method parameters.
+  !!
+  !! @author Mark Iredell, Kyle Gerheiser, Eric Engle
```

Comment:
Since you are the one adding this function, only your name should appear as author. It's understood that the code derives from other existing code, so you don't have to explicitly credit Mark and Kyle as authors. Same for any other functions you are adding...